src/data/finset/noncomm_prod.lean

```diff
@@ -244,4 +280,39 @@ begin
         list.nodup_append_of_nodup sl' tl' h]
 end
 
+section finite_pi
+
+@[to_additive]
+lemma noncomm_prod_single {ι : Type*} [fintype ι] [decidable_eq ι]
+  {M : ι → Type*} [∀ i, monoid (M i)] (x : Π i, M i) :
+  finset.univ.noncomm_prod (λ i, monoid_hom.single M i (x i))
```

Comment:
I think this should probably be stated about `mul_single`, since that's the simp-normal form?

Reply:
Is that so? I feat it might become tedious having to deal with `mul_single` when you think of it as a group homomorphism, but that bundled information (if that’s the right term) gets thrown away.

But I’ll try.

Reply:
I think I did the simp setup for `monoid_hom.single` wrong, but I think I have fixed it. With this fix, I could use `pi.mul_single` much more widely in #11744, so looks good.